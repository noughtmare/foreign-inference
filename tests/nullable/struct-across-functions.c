#include <stdlib.h>

typedef struct {
  int * ptr;
} S;

int f(S *s) {
  return *(s->ptr);
}

void g(int * arr, S *s)
{
  s->ptr = arr;
}

