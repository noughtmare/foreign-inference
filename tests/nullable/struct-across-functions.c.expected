fromList [("%struct.S",fromList ["ptr"]),("@f",fromList ["%s"]),("@g",fromList ["%arr","%s"])]
